src/nimony/sembasics.nim

```diff
@@ -180,11 +180,32 @@ proc buildErr*(c: var SemContext; info: PackedLineInfo; msg: string; orig: Curso
       if orig.kind != DotToken:
         echo "Source: ", toString(orig, false)
       quit 1
-  c.dest.buildTree ErrT, info:
-    c.dest.addSubtree orig
-    for instFrom in items(c.instantiatedFrom):
-      c.dest.add dotToken(instFrom)
-    c.dest.add strToken(pool.strings.getOrIncl(msg), info)
+  var n = orig
+  var hasErr = false
+  if n.kind == ParLe:
+    if n.tagId == ErrT:
+      hasErr = true
+    else:
+      var nested = 0
+      while true:
+        inc n
+        if n.kind == ParRi:
+          if nested == 0: break
+          dec nested
+        elif n.kind == ParLe:
+          if n.tagId == ErrT:
+            hasErr = true
+            break
+          else:
+            inc nested
+  if hasErr:
+    c.dest.addSubtree n
+  else:
+    c.dest.buildTree ErrT, info:
+      c.dest.addSubtree orig
+      for instFrom in items(c.instantiatedFrom):
+        c.dest.add dotToken(instFrom)
+      c.dest.add strToken(pool.strings.getOrIncl(msg), info)
```

Comment:
```suggestion
  c.dest.buildTree ErrT, info:
    c.dest.addSubtree orig
    for instFrom in items(c.instantiatedFrom):
      c.dest.add dotToken(instFrom)
    if hasErr:
      inc n
      while n.kind != ParRi: c.dest.takeTree n
    else:
      c.dest.add strToken(pool.strings.getOrIncl(msg), info)
```

This way the instantiation info does not get lost.

Reply:
When `hasErr` is true, it adds the content of `n` after current instantiation info. That doesn't match the existing err tag node format.
Is this means when err tag was found in `orig`, `buildErr` proc uses `orig` and the error message in `n` but uses instantiation info in the current `SemContext`?

Reply:
That's the idea, yes.